### PR TITLE
Feat: support HTTPRoute to generate endpoint

### PR DESCRIPTION
**Pull request by barnettZQG**
Signed-off-by: barnettZQG <barnett.zqg@gmail.com>

### Description of your changes

1. Support HTTPRoute to generate endpoint.
2. Remove the ExternalName option from the webservice definition, because this option can not work.
3. Add a new field `inner` for the endpoint, the endpoint is only accessible within the cluster.

I have:

- [x] Read and followed KubeVela's [contribution process](https://github.com/kubevela/kubevela/blob/master/contribute/create-pull-request.md).
- [ ] [Related Docs](https://github.com/kubevela/kubevela.io) updated properly. In a new feature or configuration option, an update to the documentation is necessary. 
- [x] Run `make reviewable` to ensure this PR is ready for review.
- [ ] Added `backport release-x.y` labels to auto-backport this PR if necessary.

### Special notes for your reviewer

@wonderflow

### CODE REVIEW

**Review by wonderflow — APPROVED** [review]

File: charts/vela-core/templates/defwithtemplate/webservice.yaml [hunk 1]
```diff
@@ -329,8 +329,8 @@ spec:
         	}]
 
         	// +ignore
-        	// +usage=Specify what kind of Service you want. options: "ClusterIP", "NodePort", "LoadBalancer", "ExternalName"
-        	exposeType: *"ClusterIP" | "NodePort" | "LoadBalancer" | "ExternalName"
```
Comment (FogDong) [inline charts/vela-core/templates/defwithtemplate/webservice.yaml:333]: Why delete the ExternalName here?

File: pkg/velaql/providers/query/endpoint.go [hunk 2]
```diff
@@ -299,6 +324,87 @@ func generatorFromIngress(ingress networkv1beta1.Ingress, cluster, component str
...
+				log.Logger.Errorf("query the Gateway %s/%s/%s failure %s", cluster, namespace, string(parent.Name), err.Error())
+			}
+			for _, listener := range gateway.Spec.Listeners {
+				if listener.Name == *parent.SectionName {
+					var protocol = querytypes.HTTP
+					if listener.Protocol == gatewayv1alpha2.HTTPSProtocolType {
+						protocol = querytypes.HTTPS
+					}
+					var port = int(listener.Port)
+					// The gateway listener port may be is not the externally exposed port.
```
Comment (FogDong) [inline pkg/velaql/providers/query/endpoint.go:345]:
```suggestion
					// The gateway listener port may not be the externally exposed port.
```

File: pkg/velaql/providers/query/endpoint.go [hunk 3]
```diff
@@ -299,6 +324,87 @@ func generatorFromIngress(ingress networkv1beta1.Ingress, cluster, component str
...
+					// Example the traefik addon default configuration, 8443->443 8000->80
```
Comment (FogDong) [inline pkg/velaql/providers/query/endpoint.go:346]:
```suggestion
					// For example, the traefik addon has a default default port configuration of 8443->443 8000->80
```

File: charts/vela-core/templates/defwithtemplate/webservice.yaml (hunk 1 above)
Comment (barnettZQG) [reply charts/vela-core/templates/defwithtemplate/webservice.yaml:333]: This option can not work, miss a required parameter.

**Review by FogDong — APPROVED** [review]

**Review by Somefive — APPROVED** [review]

**Pull request merged**